visualization/app/codeCharta/util/algorithm/treeMapLayout/modifiedTreeMapHelperFunctions.ts

```diff
@@ -0,0 +1,46 @@
+import { isLeaf } from "../../codeMapHelper"
+import { getAreaValue } from "./treeMapGenerator"
+import { HierarchyNode } from "d3-hierarchy"
+import { CodeMapNode, State } from "../../../codeCharta.model"
+
+function getLeafValues(hierarchyNode: HierarchyNode<CodeMapNode>, state: State) {
+	const leafAreaValues = []
+	for (const node of hierarchyNode) {
+		if (isLeaf(node.data)) {
+			leafAreaValues.push(getAreaValue(node.data, state))
+		}
+	}
+	return leafAreaValues
+}
+
+function getNonZeroMetrics(areaValues: number[]) {
+	let index = 0
+	while (areaValues[index] === 0) {
+		index++
+	}
+	return areaValues.slice(index)
+}
```

Comment:
```suggestion
function getMetricValuesFromFirstNonZero(areaValues: number[]) {
        areaValues.sort()
	let index = 0
	while (areaValues[index] === 0) {
		index++
	}
	return areaValues.slice(index)
}
```

Reply:
Let's sort in this method to ensure right input data.

Reply:
I sorted the result of the previous method call, but I think sorting here might be better for readability, I change it then